hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/action/rollback/TestCopyOnWriteRollbackActionExecutor.java

```diff
@@ -179,18 +181,20 @@ public void testListBasedRollbackStrategy() throws Exception {
     List<HoodieRecord> records = dataGen.generateInsertsContainsAllPartitions(newCommitTime, 3);
     JavaRDD<HoodieRecord> writeRecords = jsc.parallelize(records, 1);
     JavaRDD<WriteStatus> statuses = client.upsert(writeRecords, newCommitTime);
-    Assertions.assertNoWriteErrors(statuses.collect());
+    statuses = jsc.parallelize(statuses.collect(), 1);
+    client.commit(newCommitTime, statuses, Option.empty(), COMMIT_ACTION, Collections.emptyMap(), Option.empty());
+    assertNoWriteErrors(statuses.collect());
```

Comment:
The collect will trigger redundant write again right? Can we check by reading the write status from committed metadata.

Reply:
Let's recheck all the fixes for `assertNoWriteErrors`

Reply:
nope. we already do a collect at L184 and reconstruct the RDD<WriteStatus> based on a list of WriteStatus. 
thats what I have tried to follow whereever I have fixed. 
This will ensure we don't trigger the write handles more than once.

Reply:
I can do one more pass to ensure we don't trigger the original rdd more than once.